keywords/TestServeriOS.py

```diff
@@ -0,0 +1,323 @@
+        3. The expected version will be compared with the version reported by http://<host>:<port>
+        4. Return the url of the running LiteServ
+        """
+
+        self.logfile_name = logfile_name
+        self.app_path = "{}/{}/{}".format(BINARY_DIR, self.app_dir, self.app_name)
+
+        output = subprocess.check_output([
+            "ios-deploy", "--justlaunch", "--bundle", self.app_path
+        ])
+        log_info(output)
+
+        self._wait_until_reachable(port=self.port)
+        self._verify_running()
+
+        # return "http://{}:{}".format(self.host, self.port)
+
+    def _verify_launched(self):
+        """ Poll on expected http://<host>:<port> until it is reachable
+        Assert that the response contains the expected version information
+        """
+
+        resp_obj = self._wait_until_reachable()
+        log_info(resp_obj)
+
+        if resp_obj["vendor"]["name"] != "Couchbase Lite (Objective-C)":
+            raise LiteServError("Unexpected LiteServ platform running!")
+
+        version, build = version_and_build(self.version_build)
+        expected_version = "{} (build {})".format(version, build)
+        running_version = resp_obj["vendor"]["version"]
+
+        if expected_version != running_version:
+            raise LiteServError("Expected version: {} does not match running version: {}".format(expected_version, running_version))
+
+    def stop(self):
+        """
+        1. Flush and close the logfile capturing the LiteServ output
+        """
+        self.close_app()
+        # Have to separately copy the simulator logs
+        if self.logfile_name and self.device_id:
+            home = os.environ['HOME']
+            ios_log_file = "{}/Library/Logs/CoreSimulator/{}/system.log".format(home, self.device_id)
+            copyfile(ios_log_file, self.logfile_name)
+            # Empty the simulator logs so that the next test run
+            # will only have logs for that run
+            open(ios_log_file, 'w').close()
+
+    def _verify_running(self):
+        """
+        Return true if it is running or else false
+        Verifys that the endpoint return 200 from a running service
+        """
+        try:
+            self.session.get("http://{}:{}/".format(self.host, self.port))
+        except ConnectionError:
+            # Expecting connection error if LiteServ is not running on the port
+            raise LiteServError("Did not connected to Test server app")
+
+        return True
+
+    def close_app(self):
+        cur_dir = os.path.dirname(os.path.abspath(__file__))
+        subprocess.check_output(["osascript", "{}/../utilities/sim_close_app.scpt".format(cur_dir)])
+
+    def open_app(self):
+        bundle_id = "com.couchbase.CBLTestServer-iOS"
+        # self.app_path = "/Users/sridevi.saragadam/Library/Developer/Xcode/DerivedData/CBLTestServer-iOS-ayypbvnmphhpihebbhrjusdqhzmk/Build/Products/Debug-iphonesimulator/CBLTestServer-iOS.app"
```

Comment:
remove the commented code, if not needed.

Reply:
done